packages/bot-web-ui/src/stores/run-panel-store.js

```diff
@@ -61,23 +64,26 @@ export default class RunPanelStore {
                 'Account switching is disabled while your bot is running. Please stop your bot before switching accounts.'
             )
         );
-
-        this.is_stop_button_disabled = false;
-        this.is_running = true;
-        ui.setPromptHandler(true, route_prompt_dialog.shouldNavigateAfterPrompt);
-        this.toggleDrawer(true);
-        this.run_id = `run-${Date.now()}`;
-
-        contract_card.clear();
-        this.setContractStage(contract_stages.STARTING);
-        this.dbot.runBot();
-    };
+        runInAction(() => {
+            this.is_stop_button_disabled = false;
+            this.is_running = true;
+            ui.setPromptHandler(true, route_prompt_dialog.shouldNavigateAfterPrompt);
+            this.toggleDrawer(true);
+            this.run_id = `run-${Date.now()}`;
+
+            contract_card.clear();
+            this.setContractStage(contract_stages.STARTING);
+            this.dbot.runBot();
+        });
+    }
 
     @action.bound
     onStopButtonClick() {
+        const { self_exclusion } = this.root_store;
         const { ui } = this.root_store.core;
         this.dbot.stopBot();
         ui.setPromptHandler(false);
+        self_exclusion.resetSelfExclusion();
```

Comment:
- This should only happen `onBotStopEvent` (as is already defined), there may be situations where the bot stops due to a critical error and the stop button would never be clicked. 
- There's no need to add an additional call to `resetSelfExclusion` `onStopButtonClick` since this will eventually emit a `bot.stop`.

Reply:
> * This should only happen `onBotStopEvent` (as is already defined), there may be situations where the bot stops due to a critical error and the stop button would never be clicked.
> * There's no need to add an additional call to `resetSelfExclusion` `onStopButtonClick` since this will eventually emit a `bot.stop`.

@aaron-binary  I think it's necessary, consider a user run a bot and before his/her limit finish want to stop it and start other process if we don't reset this the limitations are still set and therefore the limit modal doesn't show so user have to refresh page to access limit modal

Reply:
In that case it will still emit a `bot.stop` event on `onStopButtonClick` because it eventually calls `terminateSession` in the interpreter.